### DISCUSSION TITLE
Next13 How does the client component use the data from the server component

### DISCUSSION THREAD

**Discussion by coderlfm** [vercel/next.js#43038] [Help]
This is how it was used in next12
```js
export async function getStaticProps() {
  const posts = await getPosts()
  return { props: { posts } }
}

function Posts(props) {
  const { data } = useQuery({
    queryKey: ['posts'],
    queryFn: getPosts,
    initialData: props.posts,
  })

  // ...
}
```
I have similar needs in next13

I want the server to have the data ready when the page first opens. For example, the first page of a blog is rendered by the server, and when the second page is clicked, the client resends the request

As far as I know, in next13 the server component couldn't transfer data to the client.

How is this requirement handled in next13?

**Comment by icyJoseph** [comment]
You just fetch in the server component whatever you need, and pass it as props to its children, one of which could potentially be a client component. https://github.com/icyJoseph/next-js-react-gbg/blob/next-13-css-modules/demo-app/app/collection/page.tsx Here you can see a server component passing data to a client children. I think the situation is somewhat similar to yours. 

An important caveat is that much like the coupling with GSP and GSSP, the props ought to be serializable.

In a sense, once the `use client` directive kicks in you tell Next.js that you take over and you can invoke react-query or whatever after that point (roughly speaking). Though I wouldn't be surprised to find bugs or that things do not straight up work at all, do remember that we are in a beta stage where all of these details and integrations are being resolved.
